syne_tune/optimizer/schedulers/searchers/bayesopt/tuning_algorithms/bo_algorithm_components.py

```diff
@@ -140,6 +140,12 @@ def f_df(x):
 
 
 class NoOptimization(LocalOptimizer):
+    def __init__(self, *args, **kwargs):
```

Comment:
Can you explain why this is needed?

Reply:
The higher level class `LocalOptimizer` needs the active metric specified and expectes to ony work for this one metric, e.g. checks if the metric if there exists a predictor for this metric. We want to re-use the same NoOptimization class for all MO metrics which does not work if the default __init__ is used. We could alternatively refactor how the NoOptimization is specificed in the BO algorithm but I thought this should work for now

Reply:
Is this something we can refactor in the base class?

Reply:
I think its good the base class checks the metric it wants ot optimize is present, it helps catch errors at construction of the optimizer. The NoOptimization dos not need any construction so its ok. I think the fix would be to adapt th BaesianOptimization constructor (create_internal) to assign optimization primitives separately to every metric. I am happy to look into it after this PR but did not want to ballon this change.